### PR TITLE
ipywidget 8 update

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: js/src/Figure.ts [hunk 1]
```diff
@@ -750,14 +754,18 @@ export class Figure extends DOMWidgetView {
...
     switch (msg.type) {
-      case 'resize':
       case 'after-show':
-      case 'after-attach':
-        if (this.pWidget.isVisible) {
+        this.shown = true;
+        this.renderImpl();
+        case 'after-attach':
+        this.attached = true;
+        this.renderImpl();
```
Comment (maartenbreddels) [inline js/src/Figure.ts:766]:
```suggestion
        this.renderImpl();
        break;
```

File: js/src/Figure.ts [hunk 2]
```diff
@@ -750,14 +754,18 @@ export class Figure extends DOMWidgetView {
...
+  processLuminoMessage(msg) {
+    super.processLuminoMessage.apply(this, arguments);
 
     switch (msg.type) {
-      case 'resize':
       case 'after-show':
-      case 'after-attach':
-        if (this.pWidget.isVisible) {
+        this.shown = true;
+        this.renderImpl();
```
Comment (maartenbreddels) [inline js/src/Figure.ts:763]:
```suggestion
        this.renderImpl();
        break;
```

File: js/src/Figure.ts [hunk 3]
```diff
@@ -100,14 +100,18 @@ export class Figure extends DOMWidgetView {
...
-    // we cannot use Promise.all here, since this.layoutPromise is resolved, and will be overwritten later on
-    this.displayed.then(() => {
-      // make sure we render after all layouts styles are set, since they can affect the size
-      this.layoutPromise.then(this.renderImpl.bind(this));
-    });
+    // We don't do anything, we'll wait for the widget to actually be visible
   }
 
   protected async renderImpl() {
+    if (!(this.attached && this.shown) || this.rendered) {
```
Comment (maartenbreddels) [inline js/src/Figure.ts:107]:
```suggestion
    if (!this.attached || !this.shown || this.rendered) {
```
equivalent using demorgan's law, and easier on my brain :)

File: js/package.json [hunk 4]
```diff
@@ -70,12 +70,12 @@
...
   },
   "dependencies": {
-    "@jupyter-widgets/base": "^2 || ^3 || ^4.0.0",
+    "@jupyter-widgets/base": "^5.0.0-beta.1",
     "@lumino/messaging": "^1",
     "@lumino/widgets": "^1",
     "@typescript-eslint/eslint-plugin": "^4.19.0",
     "@typescript-eslint/parser": "^4.19.0",
-    "bqscales": "^0.2.2",
+    "bqscales": "/home/martin/github/bqscales",
```
Comment (martinRenou) [inline js/package.json:78]: Of course, we shouldn't merge this

**Comment by martinRenou** [comment]
Seems to be working nicely

bqplot

**Comment by martinRenou** [comment]
Replaced by https://github.com/bqplot/bqplot/pull/1514

**Pull request closed without merging**